tests/unit/jobserver/test_auth_pipeline.py

```diff
@@ -1,22 +1,45 @@
-from jobserver.auth_pipeline import (
-    notify_on_new_user,
-    set_fullname,
-    set_notifications_email,
-)
-
-from ...factories import UserFactory
-
-
-def test_notify_on_new_user_with_existing_user(slack_messages):
-    user = UserFactory()
-    notify_on_new_user(user, is_new=False)
-    assert len(slack_messages) == 0
-
-
-def test_notify_on_new_user_with_new_user(slack_messages):
-    user = UserFactory()
+from jobserver.auth_pipeline import pipeline
+
+from ...factories import UserFactory, UserSocialAuthFactory
+
+
```

Comment:
Are there any failure modes that we should check for in the tests? For example, what happens if the atomic transaction fails?